File: src/jaslib_src/jaslib/ui/web/sentence/sentence_viewmodel.py
from __future__ import annotations

from typing import TYPE_CHECKING

from autoslot import Slots
from jaslib.ui.web.sentence.text_analysis_viewmodel import TextAnalysisViewModel

if TYPE_CHECKING:
    from jaslib.note.sentences.sentencenote import SentenceNote
    from jaslib.ui.web.sentence.match_viewmodel import MatchViewModel

class SentenceViewModel(Slots):
    def __init__(self, sentence: SentenceNote) -> None:
        self.sentence: SentenceNote = sentence
        self.analysis: TextAnalysisViewModel = TextAnalysisViewModel(sentence.create_analysis(for_ui=True))
        self.displayed_matches:list[MatchViewModel] = self.analysis.displayed_matches

File: src/src_dotnet/JAStudio.Core/UI/Web/Sentence/SentenceViewModel.cs
using System.Collections.Generic;
using JAStudio.Core.Configuration;
using JAStudio.Core.Note;
using JAStudio.Core.Note.Collection;
using JAStudio.Core.Note.Sentences;

namespace JAStudio.Core.UI.Web.Sentence;

public class SentenceViewModel
{
    public SentenceNote Sentence { get; }
    public TextAnalysisViewModel Analysis { get; }
    public List<MatchViewModel> DisplayedMatches { get; }

    public SentenceViewModel(SentenceNote sentence, Settings settings, VocabCollection vocab)
    {
        Sentence = sentence;
        Analysis = new TextAnalysisViewModel(sentence.CreateAnalysis(forUI: true), settings, vocab);
        DisplayedMatches = Analysis.DisplayedMatches;
    }
}
